splipy/SplineObject.py

```diff
@@ -1147,6 +1147,49 @@ def split(self, knots, direction=0):
 
         return results
 
+    def make_periodic(self, continuity=None, direction=0):
+        """Make the spline object periodic in a given parametric direction.
+
+        :param continuity: The continuity along the boundary (default max).
+        :param direction: The direction to ensure continuity in.
+        """
+
+        direction = check_direction(direction, self.pardim)
+        basis = self.bases[direction]
+        if continuity is None:
+            continuity = basis.order - 2
+        if not 0 <= continuity <= basis.order - 2:
```

Comment:
For input continuity=-1, then the right function is `SplineObject.split( )`. Might consider calling it directly from here or provide a gentle hint (error/warning message) as to where the user can find it.